Drop-In/src/test/java/com/braintreepayments/api/DropInActivityUnitTest.java

```diff
@@ -85,6 +87,153 @@ public void onCreate_whenAuthorizationIsInvalid_finishesWithError() {
         assertEquals("Tokenization Key or Client Token was invalid.", exception.getMessage());
     }
 
+    @Test
+    public void onResume_deliversBrowserSwitchResult() {
+        String authorization = Fixtures.TOKENIZATION_KEY;
+        DropInRequest dropInRequest = new DropInRequest().tokenizationKey(authorization);
+
+        DropInClient dropInClient = mock(DropInClient.class);
+
+        setupDropInActivity(authorization, dropInClient, dropInRequest, "sessionId");
+        mActivityController.setup();
+
+        verify(dropInClient).deliverBrowserSwitchResult(same(mActivity), any(DropInResultCallback.class));
+    }
+
+    @Test
+    public void onResume_whenBrowserSwitchResultExists_finishesWithResult() {
+        String authorization = Fixtures.TOKENIZATION_KEY;
+        DropInRequest dropInRequest = new DropInRequest().tokenizationKey(authorization);
+
+        DropInResult result = mock(DropInResult.class);
+        PaymentMethodNonce nonce = mock(PaymentMethodNonce.class);
+        when(result.getPaymentMethodNonce()).thenReturn(nonce);
+        when(result.getDeviceData()).thenReturn("device data");
+
+        DropInClient dropInClient = new MockDropInClientBuilder()
+                .handleThreeDSecureActivityResultSuccess(result)
+                .build();
+
+        setupDropInActivity(authorization, dropInClient, dropInRequest, "sessionId");
+        mActivityController.setup();
+
+        assertFalse(mActivity.isFinishing());
+        // TODO assert finished with result
```

Comment:
How should we approach verifying finish with result is called (when `mActivity` is not a mock)?

Reply:
I think Robolectric might allow us to inspect the activity result we pass back to the Merchant. I forget exactly how.

Reply:
Ah good call will look into that 👍